crates/client/eth/src/l1_gas_price.rs

```diff
@@ -0,0 +1,235 @@
+use crate::client::EthereumClient;
+use alloy::eips::BlockNumberOrTag;
+use alloy::providers::Provider;
+use anyhow::Context;
+use dc_mempool::L1DataProvider;
+use futures::lock::Mutex;
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::time::sleep;
+
+use lazy_static;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+const DEFAULT_GAS_PRICE_POLL_MS: u64 = 10_000;
+
+lazy_static::lazy_static! {
+    static ref LAST_UPDATE_TIMESTAMP: Arc<Mutex<u128>> = Arc::new(Mutex::new(0));
+}
+
+// Function to update the last update timestamp
+pub async fn update_last_update_timestamp() {
+    let now = SystemTime::now().duration_since(UNIX_EPOCH).expect("Time went backwards").as_millis();
+    let mut timestamp = LAST_UPDATE_TIMESTAMP.lock().await;
+    *timestamp = now;
+}
+
+// Function to get the last update timestamp
+pub async fn get_last_update_timestamp() -> u128 {
+    *LAST_UPDATE_TIMESTAMP.lock().await
+}
+
+pub async fn gas_price_worker(
+    eth_client: &EthereumClient,
+    l1_data_provider: Arc<dyn L1DataProvider>,
+    infinite_loop: bool,
+) -> anyhow::Result<()> {
+    let poll_time = eth_client.gas_price_poll_ms.unwrap_or(DEFAULT_GAS_PRICE_POLL_MS);
+    update_last_update_timestamp().await;
```

Comment:
is this incorrect? i can see the timestamp is updated inside `update_gas_price` and it should only be there from what I understand

Reply:
this is for the first time, time stamp is defined in the lazy static, and when the `gas worker` is called, it will update it for the first time and then for the next time it will get updated in the `update_gas_price`

Reply:
this is removed because we are calling the update_gas_price while creating the sync service